pkg/sql/isession/isession_test: TestInternalSession failed

pkg/sql/isession/isession_test.TestInternalSession [failed](https://mesolite.cluster.engflow.com/invocations/default/8e8d5812-41af-45d9-877e-3abcc7fa93a6?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2lzZXNzaW9uOmlzZXNzaW9uX3Rlc3Q=) on release-26.1 @ [e39600cb891be2624d33bc7d171ada4629abdfdb](https://github.com/cockroachdb/cockroach/commits/e39600cb891be2624d33bc7d171ada4629abdfdb):


```
=== RUN   TestInternalSession
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestInternalSession3497033949
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:182: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    internal_session_test.go:258: -- test log scope end --
test logs left over in: outputs.zip/logTestInternalSession3497033949
--- FAIL: TestInternalSession (18.74s)
=== RUN   TestInternalSession/tcCancelQueryBlockedOnSQLLock
    internal_session_test.go:352: 
        	Error Trace:	pkg/sql/isession/internal_session_test.go:352
        	            				pkg/sql/isession/internal_session_test.go:249
        	Error:      	An error is expected but got nil.
        	Test:       	TestInternalSession/tcCancelQueryBlockedOnSQLLock
--- FAIL: TestInternalSession/tcCancelQueryBlockedOnSQLLock (1.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInternalSession.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58995

--- a/pkg/sql/isession/internal_session_test.go
+++ b/pkg/sql/isession/internal_session_test.go
@@ -213,6 +213,11 @@ func TestInternalSession(t *testing.T) {
 	s := serverutils.StartServerOnly(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(ctx)
 
+	hostRunner := sqlutils.MakeSQLRunner(s.SystemLayer().SQLConn(t))
+	// If we have buffered writes enabled, we must have the split lock
+	// reliability enabled (which can be tweaked metamorphically, #161614).
+	hostRunner.Exec(t, "SET CLUSTER SETTING kv.lock_table.unreplicated_lock_reliability.split.enabled = true")
+
 	// This is a collection of tests that should all leave the internal session
 	// in a valid state. The order is randomized to ensure there is no accidental
 	// dependency on the order of operations. If the test fails, it is likely that

